LLDP module definition for Enterprise SONIC

divya-balasubramania opened the pull request:
LLDP module definition for Enterprise SONIC

kerry-meyer reviewed (changes requested):
Hi Divya,

This is a good base for the global LLDP model. I have posted several requests for modifications to be done before we merge it or use it as a base for the resource module.

Please take a look and let me know if you have any questions or additional considerations that I may have missed in posting my comments.

                    Kerry

divya-balasubramania commented on models/enterprise_sonic/lldp_global/deleted_example_01.txt:
```diff
@@ -0,0 +1,34 @@
+        enable: false
+        mode: receive 
+        tlv_select : 
+          'system_capabilities' : false 
```
tlv_select system_capabilities are configured by default in the system. Hence I have updated deleted_example_03.txt with the deletion of such lldp config parameters.

kerry-meyer reviewed (changes requested):
Most of the initial comments have been addressed, but I have posted some follow-up comments.

kerry-meyer commented on models/enterprise_sonic/lldp_global/deleted_example_03.txt:
```diff
@@ -0,0 +1,22 @@
+  - name: Delete default LLDP configurations
+    dellemc.enterprise_sonic.sonic_lldp_global:
+      config:
+        tlv_select : 
```
Delete the space between tlv_select and the ":".

kerry-meyer reviewed (changes requested):
A couple of the previous comments were not fully covered in the most recent push. I have posted follow-up comments requesting those minor additional changes.

There is one other change that is needed: There is no "delete all" example shown. This should be provided.

Please add a "deleted_example_04.txt" file for this purpose.

- Show "before" configuration that matches the current "after" configuration for "merged_example_01.txt" or "merged_example_02.txt".
- For the "delete_all" playbook example, do the following:

```
- name: Delete all LLDP configuration
    dellemc.enterprise_sonic.sonic_lldp_global:
      config:
      state: deleted
```

- For the "after" state, show that the output for "show running-configuration | grep lldp" is empty :

```
# sonic# show running-configuration | grep lldp
# sonic#
```

divya-balasubramania commented:
> A couple of the previous comments were not fully covered in the most recent push. I have posted follow-up comments requesting those minor additional changes.
> 
> There is one other change that is needed: There is no "delete all" example shown. This should be provided.
> 
> Please add a "deleted_example_04.txt" file for this purpose.
> 
> * Show "before" configuration that matches the current "after" configuration for "merged_example_01.txt" or "merged_example_02.txt".
> * For the "delete_all" playbook example, do the following:
> 
> ```
> - name: Delete all LLDP configuration
>   dellemc.enterprise_sonic.sonic_lldp_global:
>       config:
>       state: deleted
> ```
> 
> * For the "after" state, show that the output for "show running-configuration | grep lldp" is empty :
> 
> ```
> # sonic# show running-configuration | grep lldp
> # sonic#
> ```

deleted_example_02,txt has delete all configs. --> modified now as deleted_example_03.txt
Updated the empty configs as requested.

kerry-meyer commented on models/enterprise_sonic/lldp_global/merged_example_01.txt:
```diff
@@ -0,0 +1,35 @@
+              hello_time: 18
+              mode: receive
+              system_description: Sonic_System
+              suppress_tlv_select: {
```
Sorry I missed this earlier:

- The name that you have given this item in the model is "tlv_select" (not "suppress_tlv_select"). Please fix that here.
- Delete the open and close braces and the quotes around the attribute names. This is not correct for multi-line YAML dict definitions. (REF: https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html).

divya-balasubramania replied:
addressed.

kerry-meyer reviewed (changes requested):
The most recent comments have been addressed as requested.

In doing a final scan of the change set, I found a couple of other mistakes added with intermediate commits when making other code review changes.

Please fix these as noted in the comment I am posting now.

Other than the lines noted in this new comment, the rest of the change set looks good now.

kerry-meyer reviewed (approved):
All remaining comments have been addressed.

This change set looks good for merging.

Approved.
